catalog/async/catalog-async-data/src/test/groovy/org/codice/ddf/catalog/async/data/LazyProcessResourceImplTest.groovy

```diff
@@ -0,0 +1,175 @@
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.async.data
+
+import ddf.catalog.resource.Resource
+import org.codice.ddf.catalog.async.data.impl.LazyProcessResourceImpl
+import spock.lang.Shared
+import spock.lang.Specification
+
+import java.util.function.Supplier
+
+import static org.codice.ddf.catalog.async.data.impl.ProcessResourceImpl.*
+
+class LazyProcessResourceImplTest extends Specification {
+
+    @Shared
+    def RESOURCE_NAME = "resourceName"
+
+    @Shared
+    int RESOURCE_SIZE = 1
+
+    @Shared
+    InputStream RESOURCE_INPUTSTREAM = Mock(InputStream)
+
+    @Shared
+    String RESOURCE_MIMETYPE = "mimeType"
+
+    @Shared
+    URI RESOURCE_URI = new URI(CONTENT_SCHEME, "metacardId", null)
+
+    @Shared
+    String RESOURCE_QUALIFIER = ""
+
+
+    private Resource resource
+    private Supplier<Resource> supplier
+
+    def setup() {
+        resource = Stub(Resource) {
+            getSize() >> RESOURCE_SIZE
+            getInputStream() >> RESOURCE_INPUTSTREAM
+            getName() >> RESOURCE_NAME
+            getMimeTypeValue() >> RESOURCE_MIMETYPE
+        } as Resource
+
+        supplier = Mock(Supplier) {
+            get() >> resource
+        } as Supplier<Resource>
+    }
+
+
+    def 'new resource IllegalArgument on bad inputs'() {
+        when:
+        new LazyProcessResourceImpl(metacardId, resourceSupplier)
+
+        then:
+        thrown(IllegalArgumentException)
+
+        where:
+        metacardId   | resourceSupplier
+        null         | Mock(Supplier)
+        ""           | Mock(Supplier)
+        " "          | Mock(Supplier)
+        "metacardId" | null
+
+    }
+
+    def 'new resource verifying lazy loading not loaded'() {
+        when:
+        new LazyProcessResourceImpl("metacardId", supplier)
+
+        then:
+        0 * supplier.get()
+    }
+
+    def 'new resource verifying lazy loaded at first get'() {
+        given:
+        Supplier<Resource> supplier = Mock(Supplier) {
+            1 * get() >> resource
+        }
+
+        def lazyProcessResource = new LazyProcessResourceImpl("metacardId", supplier)
+
+        when:
+        lazyProcessResource.getSize()
+        lazyProcessResource.getName()
+        lazyProcessResource.close()
+
+        then:
+        lazyProcessResource.size == RESOURCE_SIZE
+        lazyProcessResource.name == RESOURCE_NAME
+        lazyProcessResource.inputStream == RESOURCE_INPUTSTREAM
+        lazyProcessResource.mimeType == RESOURCE_MIMETYPE
+        lazyProcessResource.qualifier == RESOURCE_QUALIFIER
+        lazyProcessResource.uri == RESOURCE_URI
+        !lazyProcessResource.modified
+    }
+
+    def 'new resource verify not loaded on close'() {
+        given:
+        def lazyProcessResource = new LazyProcessResourceImpl("metacardId", supplier)
+
+        when:
+        lazyProcessResource.close()
+
+        then:
+        0 * supplier.get()
```

Comment:
✏️ Can we also verify that close is not called on the `InputStream` since we have not loaded it?